src/storage/irc_table_entry.cpp

```diff
@@ -86,42 +72,37 @@ TableFunction ICTableEntry::GetScanFunction(ClientContext &context, unique_ptr<F
-		info.name = "__internal_ic_" + table_data->table_id + "__" + table_data->schema_name + "__" + table_data->name;
-		info.type = "s3";
-		info.provider = "config";
-		info.storage_type = "memory";
-
-		auto substrings = StringUtil::Split(ic_catalog.warehouse, ":");
-		D_ASSERT(substrings.size() == 6);
-		auto region = substrings[3];
-		auto endpoint = "s3." + region + ".amazonaws.com";
-		info.options = {{"key_id", kv_secret.TryGetValue("key_id").ToString()},
-		                {"secret", kv_secret.TryGetValue("secret").ToString()},
-		                {"session_token", kv_secret.TryGetValue("session_token").IsNull()
-		                                      ? ""
-		                                      : kv_secret.TryGetValue("session_token").ToString()},
-		                {"region", region},
-		                {"endpoint", endpoint}};
 
-		std::string lc_storage_location;
-		lc_storage_location.resize(table_data->storage_location.size());
-		std::transform(table_data->storage_location.begin(), table_data->storage_location.end(),
-		               lc_storage_location.begin(), ::tolower);
-		size_t metadata_pos = lc_storage_location.find("metadata");
-		if (metadata_pos != std::string::npos) {
-			info.scope = {lc_storage_location.substr(0, metadata_pos)};
-		} else {
-			throw std::runtime_error("Substring not found");
+		if (StringUtil::StartsWith(ic_catalog.host, "glue")) {
+			//! Override the endpoint if 'glue' is the host of the catalog
+			auto secret_entry = IRCatalog::GetSecret(context, ic_catalog.secret_name);
+			auto kv_secret = dynamic_cast<const KeyValueSecret &>(*secret_entry->secret);
+			auto region = kv_secret.TryGetValue("region").ToString();
+			auto endpoint = "s3." + region + ".amazonaws.com";
+			info.options["endpoint"] = endpoint;
+		} else if (StringUtil::StartsWith(ic_catalog.host, "s3tables")) {
+			//! Override all the options if 's3tables' is the host of the catalog
+			auto secret_entry = IRCatalog::GetSecret(context, ic_catalog.secret_name);
+			auto kv_secret = dynamic_cast<const KeyValueSecret &>(*secret_entry->secret);
+			auto substrings = StringUtil::Split(ic_catalog.warehouse, ":");
+			D_ASSERT(substrings.size() == 6);
+			auto region = substrings[3];
+			auto endpoint = "s3." + region + ".amazonaws.com";
+			info.options = {{"key_id", kv_secret.TryGetValue("key_id").ToString()},
+			                {"secret", kv_secret.TryGetValue("secret").ToString()},
+			                {"session_token", kv_secret.TryGetValue("session_token").IsNull()
+			                                      ? ""
+			                                      : kv_secret.TryGetValue("session_token").ToString()},
+			                {"region", region},
+			                {"endpoint", endpoint}};
 		}
 		auto my_secret = secret_manager.CreateSecret(context, info);
 	}
 
+	for (auto &info : table_credentials.storage_credentials) {
+		auto my_secret = secret_manager.CreateSecret(context, info);
```

Comment:
nit: I think you can remove the `auto my_secret =` if you want (and there are no compilation errors etc.). The variable isn't used.